org.lflang/src/org/lflang/federated/generator/FedGenerator.java

```diff
@@ -321,7 +330,7 @@ public TargetConfig getTargetConfig() {
     try {
       compileThreadPool.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
     } catch (Exception e) {
-      Exceptions.sneakyThrow(e);
+      context.getErrorReporter().reportError("Failure during code generation.");
```

Comment:
To my knowledge, it "should be impossible" for errors to bubble up to this point, in the sense that if there is an exception here, it indicates that the compiler has a bug, and that the problem is not merely a matter of invalid user input.

If that is the case, then I think that we should print the full stack trace of the error message and/or re-throw it as a `RuntimeException`. The resulting stack trace will make the output of the compiler look bad, but if the compiler has a bug then it should admit that by allowing its output to look bad. This will be more helpful for debugging than an error message like "Failure during code generation," which is not very specific and which does not provide any hints about where in the code to look for the error.